### PR TITLE
fix(Modal): add keyboard trap

### PR DESCRIPTION
**Pull request by Kyle-Cooley**
Adds a keyboard trap to fix issue #874

### CODE REVIEW

**Comment by Kyle-Cooley** [comment]
Hello @asudoh. That was a good catch. I have amended my commit to check if the modal is open when trapping focus.

**Comment by asudoh** [comment]
Hi @Kyle-Cooley thank you for your quick change! Just a couple of additional things:

1. Vanilla version moves the focus (back) to modal itself, instead of preventing the focus from moving. Can we align the behaviors?
2. Vanilla version makes focus to floating menu an exception, due to https://github.com/IBM/carbon-components/issues/280. Can we address this in this React variant, too?

FYI here's the corresponding code in vanilla: https://github.com/IBM/carbon-components/blob/3e9bc26/src/components/modal/modal.js#L140-L153

**Comment by Kyle-Cooley** [comment]
I addressed the first, but I'm struggling with the second. I'm using an overflow menu (floating) in a modal. The items in the menu can be tabbed to, but when the overflow menu closes, focus is not returned to the modal.

**Comment by Kyle-Cooley** [comment]
I think I've managed to address the issue, let me know if there is something else.

File: src/components/OverflowMenu/OverflowMenu.js [hunk 1]
```diff
@@ -323,7 +323,11 @@ export default class OverflowMenu extends Component {
   };
 
   closeMenu = () => {
+    let wasOpen = this.state.open;
     this.setState({ open: false }, () => {
+      if (wasOpen) {
+        this.focusMenuEl();
```
Comment (asudoh) [inline src/components/OverflowMenu/OverflowMenu.js]: Did you want to remove: https://github.com/Kyle-Cooley/carbon-components-react/blob/742a24bb891752975242ccc388f49f22515c0948/src/components/OverflowMenu/OverflowMenu.js#L383-L386

File: src/components/Modal/Modal.js [hunk 2]
```diff
@@ -36,10 +43,21 @@ export default class Modal extends Component {
     iconDescription: 'close the modal',
     modalHeading: '',
     modalLabel: '',
+    selectorsFloatingMenus: ['.bx--overflow-menu-options__btn'],
```
Comment (asudoh) [inline src/components/Modal/Modal.js]: Do you think it's possible to align the list with vanilla? https://github.com/IBM/carbon-components/blob/3e9bc26/src/components/modal/modal.js#L194

**Comment by joshblack** [comment]
Hi @Kyle-Cooley! Looks great, mind fixing the merge conflicts and then we can merge?

**Comment by Kyle-Cooley** [comment]
Should be good now.

**Pull request merged**